build.gradle

```diff
@@ -208,90 +203,39 @@ task setupEclipse {
-
-task addAppEngineSdkHomePath(dependsOn: appengineDownloadSdk) {
-    group intellijSetupGroup
-    doLast {
-        String moduleXmlFilePath = '.idea/modules/teammates.iml'
-        Node moduleXml = new XmlParser().parse(moduleXmlFilePath)
-        Node facetComponent = (Node) (moduleXml.component.find { it.@name == 'FacetManager' })
-        Node googleAppEngineFacet = (Node) (facetComponent.find { it.@type == 'google-app-engine' })
-        Node configuration = googleAppEngineFacet.configuration[0]
-
-        NodeList sdkHomePathList = configuration.'sdk-home-path'
-        if (sdkHomePathList.size() > 1) {
-            throw new GradleException('More than one sdk-home-path found in Google App Engine Facet')
-        } else if (sdkHomePathList.size() == 1) {
-            configuration.'sdk-home-path'[0].value = System.properties['appengine.sdk.root']
-        } else {
-            List configurationChildren = configuration.children()
-            configurationChildren.add(0, new Node(null, 'sdk-home-path', null, System.properties['appengine.sdk.root']))
-        }
-
-        new File(moduleXmlFilePath).withWriter('UTF-8') { out ->
-            out << new StreamingMarkupBuilder().bind { mkp.xmlDeclaration() }
-            def xmlNodePrinter = new XmlNodePrinter(new PrintWriter(out))
-            xmlNodePrinter.preserveWhitespace = true
-            xmlNodePrinter.print(moduleXml)
-        }
-    }
-}
 
-task deleteIdeaFolder(type: Delete) {
-    delete '.idea'
+// Use a different buildDir for IntelliJ builds to avoid conflict with CLI builds
+if (System.getProperty('idea.executable') != null) {
+    // Nest the buildDir inside IntelliJ's default project compiler output directory
+    buildDir = new File(projectDir, 'out/build')
 }
 
-task copyIntelliJProjectSettings(type: Copy) {
-    group intellijSetupGroup
-    if (file('.idea').exists()) {
-        if (!file('.idea').isDirectory()) {
-            throw new GradleException('Found file with name .idea where it should be a folder.' +
-                    'Please rename or delete it otherwise setup cannot proceed.')
-        }
-
-        if (project.hasProperty('delete_previous_idea')) {
-            println 'Your previous project will be overwritten.'
-            dependsOn deleteIdeaFolder
-        } else {
-            doLast {
-                throw new GradleException('A project folder is found, you can delete it manually or ' +
-                        'call the setup again with the property -Pdelete_previous_idea')
-            }
-        }
+idea {
+    module {
+        // Exclude the CLI and Eclipse build directories
+        excludeDirs += [file('build'), file('build-eclipse')]
     }
-    from '.templates/.idea'
-    into '.idea'
 }
 
 task setupIntellijRunConfigs(type: Copy) {
+    description 'Sets up the run configurations in the IntelliJ project.'
     group intellijSetupGroup
-    from '.templates/.idea/runConfigurations'
+    doFirst {
+        if (!file('.idea').exists()) {
```

Comment:
What's the reason for this though? Will it get overwritten if it's done before importing?

Reply:
Yes; IntelliJ will warn and the user will be forced to either cancel the import or proceed with the overwriting.